zebra-script/src/lib.rs

```diff
@@ -2,10 +2,166 @@
+    zcashconsensus_error_t_zcashconsensus_ERR_TX_DESERIALIZE,
+    zcashconsensus_error_t_zcashconsensus_ERR_TX_INDEX,
+    zcashconsensus_error_t_zcashconsensus_ERR_TX_SIZE_MISMATCH,
+};
+use zebra_chain::{
+    block::Block,
+    transaction::{TransparentInput, TransparentOutput},
+};
+
+#[derive(Debug, Display, Error)]
+#[non_exhaustive]
+pub enum Error {
+    /// ErrOk variant, what does this mean?
+    #[non_exhaustive]
+    ErrOk,
+    /// Unable to deserialize tx_to input
+    #[non_exhaustive]
+    TxDeserialize,
+    /// TxIndex
+    #[non_exhaustive]
+    TxIndex,
+    /// TxSizeMismatch
+    #[non_exhaustive]
+    TxSizeMismatch,
+    /// encountered unknown error kind from zcashconsensus: {0}
+    #[non_exhaustive]
+    Unknown(zcashconsensus_error_t),
+}
+
+impl From<zcashconsensus_error_t> for Error {
+    #[allow(non_upper_case_globals)]
+    fn from(err_code: zcashconsensus_error_t) -> Error {
+        match err_code {
+            zcashconsensus_error_t_zcashconsensus_ERR_OK => Error::ErrOk,
+            zcashconsensus_error_t_zcashconsensus_ERR_TX_DESERIALIZE => Error::TxDeserialize,
+            zcashconsensus_error_t_zcashconsensus_ERR_TX_INDEX => Error::TxIndex,
+            zcashconsensus_error_t_zcashconsensus_ERR_TX_SIZE_MISMATCH => Error::TxSizeMismatch,
+            unknown => Error::Unknown(unknown),
+        }
+    }
+}
+
+pub fn verify_script(
+    script_pub_key: impl AsRef<[u8]>,
+    amount: i64,
+    tx_to: impl AsRef<[u8]>,
+    n_in: u32,
+    flags: u32,
+    consensus_branch_id: u32,
+) -> Result<(), Error> {
+    let script_pub_key = script_pub_key.as_ref();
+    let tx_to = tx_to.as_ref();
+
+    let script_ptr = script_pub_key.as_ptr();
+    let script_len = script_pub_key.len();
+    let tx_to_ptr = tx_to.as_ptr();
+    let tx_to_len = tx_to.len();
+    let mut err = 0;
+
+    let ret = unsafe {
+        zcashconsensus::zcashconsensus_verify_script(
+            script_ptr,
+            script_len as u32,
+            amount,
+            tx_to_ptr,
+            tx_to_len as u32,
+            n_in,
+            flags,
+            consensus_branch_id,
+            &mut err,
+        )
+    };
+
+    dbg!(&ret, &err);
+
+    if ret == 1 {
+        Ok(())
+    } else {
+        Err(Error::from(err))
+    }
+}
+pub fn script_is_valid(block: &Block) -> bool {
+    for trans in &block.transactions {
+        let inputs = trans.inputs();
+        let outputs = trans.outputs();
+
+        for (input, output) in inputs.zip(outputs) {
+            let TransparentOutput { value, pk_script } = output;
+            let script = if let TransparentInput::PrevOut { script, .. } = input {
+                script
+            } else {
+                continue;
+            };
+
+            let n_in = 0;
+            let flags = 1;
+            let branch_id = 0x2bb40e61;
+
+            verify_script(pk_script, (*value).into(), script, n_in, flags, branch_id).unwrap();
+        }
+    }
+
+    true
```

Comment:
I'm extremely YOLOing this part as I try to figure out what the correct inputs are supposed to be